python/lsst/meas/extensions/gaap/_gaap.py

```diff
@@ -551,8 +551,18 @@ def _gaussianizeAndMeasure(self, measRecord: lsst.afw.table.SourceRecord,
             raise measBase.FatalAlgorithmError("No PSF in exposure")
         wcs = exposure.getWcs()
 
-        psfSigma = psf.computeShape(center).getDeterminantRadius()
-        errorCollection = dict()
+        psfShape = psf.computeShape(center)
+        if not (psfShape.getDeterminantRadius() > 0):
+            self.log.debug("PSF shape has non-positive determinant radius at (%f, %f)", center.x, center.y)
+
+        psfSigma = psfShape.getTraceRadius()
+        if not (psfSigma > 0):  # This captures nan
```

Comment:
If all you want to do here is capture NaNs, wouldn't `np.isnan` be better? Or do you expect some negative values of `getTraceRadius` to creep through for other reasons? If so, the in-line comment should reflect that possibility.

Reply:
I don't think either of `getTraceRadius` or `getDeterminantRadius` will ever return negative numbers, but in the off-chance they returned -1 or something, this will capture that. If I remember right, when I wrote this line, I didn't want to have to import numpy  just for this condition. The way I read it, if the PSF size is not a positive-valued number, then don't proceed.  Do you think `np.isnan` makes the intention much clearer?

Reply:
If there is a chance that a negative number may be returned, then yes, it makes sense to leave it as-is. I would however recommend changing the in-line comment to something like `This captures NaN or negative values`.